### PR TITLE
fix(dockerhub): dockerhub ratelimit issues EE-1035

### PR DESCRIPTION
**Pull request by hookenz**
Fixes: [EE-1035](https://portainer.atlassian.net/browse/EE-1035)

### CODE REVIEW

**Comment by zees-dev** [comment]
EE does not return error early if there are issues with rate limit responses:
https://github.com/portainer/portainer-ee/blob/f0a1bd7b036054cb560fa6414d02c8900d1c54fb/api/http/handler/endpoints/endpoint_dockerhub_status.go#L137

You may want to do the same here?

File: http/handler/dockerhub/dockerhub_status.go [hunk 1]
```diff
@@ -115,14 +115,16 @@ func getDockerHubLimits(httpClient *http.Client, token string) (*dockerhubStatus
...
 	if err != nil {
-		return nil, fmt.Errorf("Failed fetching RateLimit-Limit header: %w", err)
+		return nil, nil
 	}
 
 	rateLimitRemaining, err := parseNumericHeader(resp.Header, "RateLimit-Remaining")
 	if err != nil {
-		return nil, fmt.Errorf("Failed fetching RateLimit-Remaining header: %w", err)
+		return nil, nil
 	}
```
Comment (zees-dev) [inline http/handler/dockerhub/dockerhub_status.go:128]: Could just be:
```go
rateLimit, _ := parseNumericHeader(resp.Header, "RateLimit-Limit")
rateLimitRemaining, _ := parseNumericHeader(resp.Header, "RateLimit-Remaining")
```

**Review by zees-dev — CHANGES_REQUESTED** [review]
Would prefer suggested change to retain parity/familiarity with EE.

**Review by zees-dev — APPROVED** [review]
Discussed on chat; LGTM 👍

**Review by sbusso — APPROVED** [review]

**Pull request merged**